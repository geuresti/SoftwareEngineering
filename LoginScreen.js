import React, { useState } from 'react';
import { Text, View, ImageBackground, TouchableOpacity, TextInput, StyleSheet, Alert} from "react-native";
import { openDatabase } from 'react-native-sqlite-storage';
import dbModel from './dbModel';

var db = openDatabase({ name: 'UserDatabase.db' });

const Login = ({ navigation }) => {
  dao = new dbModel()


    const styles = StyleSheet.create({
        input: {
          borderColor: "gray",
          color: "black",
          backgroundColor: "white",
          width: "100%",
          borderWidth: 2,
          borderRadius: 10,
          padding: 10,
        },
        button: {
            //flex: 1,
            alignItems: "center",
            backgroundColor: "#171414",
            padding: 15
            
        }
      });
    /*
      useEffect(() => {
        db.transaction(function (txn) {
          txn.executeSql(
            "SELECT name FROM sqlite_master WHERE type='table' AND name='user_table'",
            [],
            function (tx, res) {
              console.log('item:', res.rows.length);
              if (res.rows.length == 0) {
                txn.executeSql('DROP TABLE IF EXISTS user_table', []);
                txn.executeSql(
                  'CREATE TABLE IF NOT EXISTS user_table(user_id INTEGER PRIMARY KEY AUTOINCREMENT, user_email VARCHAR(20), user_password VARCHAR(20))',
                  []
                );
              }
            }
          );
        });
      }, []);*/

  let [userEmail, setUserEmail] = useState('');
  let [userPassword, setUserPassword] = useState('');

  let register_user = () => {
    //console.log(userEmail, userPassword);

    if (!userEmail) {
      alert('Please fill email');
      return;
    }
    if (userPassword.length < 7) {
      alert('Password must be at least 8 characters');
      return;
    }

    dao.createUser(userEmail, userPassword)
    Alert.alert(
      'Success',
      'You are Registered Successfully',
      [
        {
          text: 'Ok',
          onPress: () => navigation.navigate('AdminPage'),
        },
      ],
      { cancelable: false }
    );
    
  };
    
    
   return (
      <View style={{flex:1, alignItems:'center', justifyContent:'center'}}>
         <ImageBackground
          style={{
            //resizeMode: "center",
            height: '100%',
            width: '100%'
          }}
          source={require("./bg.png")}
        />
        <View style={{position: 'absolute', top: 0, left: 0, right: 0, bottom: 500, justifyContent: 'center', alignItems: 'center'}}>
        <Text style={{fontSize:35 , fontFamily: 'Bungee-Regular', color: '#D9D9D9'}}>LOGIN</Text>
        </View>
        <View style={{position: 'absolute', top: 0, left: 0, right: 270, bottom: 250, justifyContent: 'center', alignItems: 'center'}}>
        <Text style={{fontSize:20 , fontFamily: 'monospace', color: '#D9D9D9'}}>Username</Text>
        </View>
        <View style={{position: 'absolute', top: 0, left: 0, right: 0, bottom: 90, justifyContent: 'center', alignItems: 'center'}}>
        <TextInput 
          style = {styles.input} keyboardType="email-address"
          textAlign={'center'}
          placeholder="Email"
          onChangeText={
            (userEmail) => setUserEmail(userEmail)
          }
          />
        </View>
        <View style={{position: 'absolute', top: 50, left: 0, right: 270, bottom: 0, justifyContent: 'center', alignItems: 'center'}}>
        <Text style={{fontSize:20 , fontFamily: 'monospace', color: '#D9D9D9'}}>Password</Text>
        </View>
        <View style={{position: 'absolute', top: 200, left: 0, right: 0, bottom: 0, justifyContent: 'center', alignItems: 'center'}}>
        <TextInput
          style ={styles.input} keyboardType="default"
          textAlign={'center'}
          secureTextEntry={true}
          placeholder="Password"
          onChangeText={
            (userPassword) => setUserPassword(userPassword)
          }
          />
        </View>
        <View style={{position: 'absolute', top: 450, left: 200, right: 0, bottom: 0, justifyContent: 'center', alignItems: 'center'}}>
        <TouchableOpacity
            //onPress={() => console.log("button pressed!")} 
            onPress={register_user}
            style={styles.button}>
        <Text style={{color: "#FFFFFF", fontFamily: 'monospace'}}>Login</Text>
        </TouchableOpacity>
  
        </View>
      </View>
    
   );

}
export default Login;
