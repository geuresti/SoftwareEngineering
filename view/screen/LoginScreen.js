import React from "react";
import { Text, View, Image, ImageBackground, TouchableOpacity, TextInput, StyleSheet, Button, Alert, SafeAreaView, FlatList} from "react-native";
import FormStyle from "../Form.style";
import { useState } from 'react';
import Realm from "realm";
//import LoginScreenAuth from "./view/screen/LoginScreenAuth"; 

/*realm = LoginRealm;  */

realm = new Realm({path: 'logins.realm',
schema:[
    {
    name: "User",
    properties: {
        username: "string",
        pass: "string",
    },
    primaryKey: "username",


    },
],

});
const email = "someone@example.com";
const password = "Pa55w0rd";
// LoginScreenAuth.loginEmailPassword(email, password)

export default function LoginScreen({navigation}){
    let [userEmail, setUserEmail] = useState('');
    let [userPassword, setUserPassword] = useState('');
    let register_user = () => {
        if (!userEmail) {
            alert('Please fill email');
            return;
          }
          /* the email password authentication provides users the ability to log in to app w/ email address and password */
          if (userEmail !="hemannering1@loyola.edu" || userPassword != "password1234") {
            alert('Not correct user, please try again');
            return;
          }
          if (userPassword.length < 8) {
            alert('Password must be at least 8 characters');
            return;
          }
          
        let user;
        
    realm.write(() => {
        user = realm.create("User", {username: userEmail, pass: userPassword});
      })

    Alert.alert(
      'Success',
      'You are Registered Successfully',
      [
        {
          text: 'Ok',
          onPress: () => navigation.navigate(AdminPage),
        },
      ],
      { cancelable: false }
    );
    
  };
    
    

    return (
              <View style={{flex:1, alignItems:'center', justifyContent:'center'}}>
              <ImageBackground
              style={{
              //resizeMode: "center",
              height: '100%',
              width: '100%'
              }}
              source={require("./bg.png")}
          />
          <View style={{position: 'absolute', top: 0, left: 0, right: 0, bottom: 500, justifyContent: 'center', alignItems: 'center'}}>
          <Text style={{fontSize:35 , fontFamily: 'Bungee-Regular', color: '#D9D9D9'}}>LOGIN</Text>
          </View>
          <View style={{position: 'absolute', top: 0, left: 0, right: 270, bottom: 250, justifyContent: 'center', alignItems: 'center'}}>
          <Text style={{fontSize:20 , fontFamily: 'monospace', color: '#D9D9D9'}}>Username</Text>
          </View>
          <View style={{position: 'absolute', top: 0, left: 0, right: 0, bottom: 90, justifyContent: 'center', alignItems: 'center'}}>
          <TextInput 
              style = {FormStyle.input} keyboardType="email-address"
              textAlign={'center'}
              placeholder="Email"
              onChangeText={
              (userEmail) => setUserEmail(userEmail)
              }
              />
          </View>
          <View style={{position: 'absolute', top: 50, left: 0, right: 270, bottom: 0, justifyContent: 'center', alignItems: 'center'}}>
          <Text style={{fontSize:20 , fontFamily: 'monospace', color: '#D9D9D9'}}>Password</Text>
          </View>
          <View style={{position: 'absolute', top: 200, left: 0, right: 0, bottom: 0, justifyContent: 'center', alignItems: 'center'}}>
          <TextInput
              style ={FormStyle.input} keyboardType="default"
              textAlign={'center'}
              secureTextEntry={true}
              placeholder="Password"
              onChangeText={
              (userPassword) => setUserPassword(userPassword)
              }
              />
          </View>
          <View style={{position: 'absolute', top: 450, left: 200, right: 0, bottom: 0, justifyContent: 'center', alignItems: 'center'}}>
          
          <TouchableOpacity
              //onPress={() => console.log("button pressed!")} 
              onPress={register_user}
              style={FormStyle.button}>
          <Text style={{color: "#FFFFFF", fontFamily: 'monospace'}}>Login</Text>
          </TouchableOpacity>
      
          </View>
          </View>
  
      
          );
    }