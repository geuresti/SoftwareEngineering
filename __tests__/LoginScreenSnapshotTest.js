import React from "react";
import renderer from "react-test-renderer";
import LoginScreen from "../LoginScreen";


    it("Login Snapshot", () => {
        const domTree = renderer.create(<LoginScreen />).toJSON();
        expect(domTree).toMatchSnapshot();
  });
  
