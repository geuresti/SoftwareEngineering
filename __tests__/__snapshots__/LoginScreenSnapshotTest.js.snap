// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../bg.png",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": "100%",
            "width": "100%",
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "bottom": 500,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#D9D9D9",
          "fontFamily": "Bungee-Regular",
          "fontSize": 35,
        }
      }
    >
      LOGIN
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "bottom": 250,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 270,
        "top": 0,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#D9D9D9",
          "fontFamily": "monospace",
          "fontSize": 20,
        }
      }
    >
      Username
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "bottom": 90,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <TextInput
      keyboardType="email-address"
      onChangeText={[Function]}
      placeholder="Email"
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "gray",
          "borderRadius": 10,
          "borderWidth": 2,
          "color": "black",
          "padding": 10,
          "width": "100%",
        }
      }
      textAlign="center"
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "bottom": 0,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 270,
        "top": 50,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#D9D9D9",
          "fontFamily": "monospace",
          "fontSize": 20,
        }
      }
    >
      Password
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "bottom": 0,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 200,
      }
    }
  >
    <TextInput
      keyboardType="default"
      onChangeText={[Function]}
      placeholder="Password"
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "gray",
          "borderRadius": 10,
          "borderWidth": 2,
          "color": "black",
          "padding": 10,
          "width": "100%",
        }
      }
      textAlign="center"
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "bottom": 0,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 200,
        "top": 450,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#171414",
          "opacity": 1,
          "padding": 15,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "monospace",
          }
        }
      >
        Login
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "bottom": 0,
        "justifyContent": "center",
        "left": 200,
        "position": "absolute",
        "right": 0,
        "top": 450,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#171414",
          "opacity": 1,
          "padding": 15,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "monospace",
          }
        }
      >
        Register
      </Text>
    </View>
  </View>
</View>
`;
